config/openshift/v4_12_exp/translate.go

```diff
@@ -295,3 +296,25 @@ func validateMCOSupport(mc result.MachineConfig, ts translate.TranslationSet) re
 	}
 	return cutil.TranslateReportPaths(r, ts)
 }
+
+// fcos v1.5.0-exp generates a user.cfg file using append; however, OpenShift config
+// does not support append (since MCO does not support it). Let change the file to use contents
+func translateUserGrubCfg(config *types.Config, ts *translate.TranslationSet) translate.TranslationSet {
+	newMappings := translate.NewTranslationSet("json", "json")
+	for i, file := range config.Storage.Files {
+		if file.Path == "/boot/grub2/user.cfg" {
+			if len(file.Append) != 1 {
+				// The number of append objects was different from expected, this file
+				// was created by the user and not via butane GRUB sugar
+				return *ts
+			}
+			translatedPath := path.New("json", "storage", "files", i, "contents")
+			config.Storage.Files[i].FileEmbedded1.Contents = file.Append[0]
+			config.Storage.Files[i].FileEmbedded1.Append = nil
+			newMappings.AddFromCommonObject(path.New("json", "storage", "files", i, "append", 0), file.Append[0], translatedPath, file.Contents)
```

Comment:
This doesn't affect the functionality because the field contents don't actually matter here, but: `file.Contents` is an empty struct.  It might be clearer to pass the populated copy of the struct.

Also, it's a bit odd that one of the paths is declared beforehand and the other one isn't.